lib/matplotlib/artist.py

```diff
@@ -34,7 +34,17 @@ def allow_rasterization(draw):
     def draw_wrapper(artist, renderer, *args, **kwargs):
         try:
             if artist.get_rasterized():
-                renderer.start_rasterizing()
+                if renderer._raster_depth == 0 and not renderer._rasterizing:
```

Comment:
I think it is ok to try and monkey-patch these attributes on if they don't exist.

It is a little bit dirty, but greatly reduces the chances we break anyone.

Reply:
This doesn't really work if the renderer class has no `__dict__` (given that renderer classes can easily be defined in extension modules this is not just a hypothetical).
I think saying "you have to inherit from RendererBase" (or are responsible for tracking changes upstream) is the reasonable way to go.

Reply:
The new requirement for those attributes is noted in the API changes. Should I add the text about inheriting from RenderBase there? or somewhere else? I would think that is it implicit that if a 3rd party replaces a class without inheriting from the original that they need to track upstream changes.

In  https://github.com/matplotlib/matplotlib/pull/17159#issuecomment-619610790 you say that mplcairo can handle the change.